backups/README.md

```diff
@@ -0,0 +1,198 @@
+
+There are a number of considerations for choosing the location type and storage class.
+We recommend reviewing Google Cloud documentation to understand the tradeoffs involved in
+making those choices. For example, a Multi-region location type would incur the highest cost
+but will provide the highest availability. A Nearline storage class could be appropriate for
+backups, but you would need to evaluate this based on cost and performance characteristics.
+
+We recommend configuring a uniform access to all objects in the bucket.
+
+Repeat the bucket creation process for the incremental and regular buckets.
+
+### Granting permissions
+
+You can grant permissions by opening bucket details and navigating to the Permissions tab.
+
+-  For the incremental bucket, grant the Storage Object Admin role to the
+   Perforce VM service account.
+-  For the regular bucket, grant the Storage Object Creator role to the
+   Perforce VM service account.
+
+Note: A GCP VM additionally needs the Access scopes to be configured to enabled writing
+to Storage (e.g. granting Read Write or Full Storage scopes).
+
+Note: If you change the settings but still get 403 when writing via gsutil, the issue can
+sometimes be fixed by removing the ~/.gsutil folder.
+
+## Configuring Perforce user
+
+The scripts assume that a dedicated Linux user has been created on the machine running Helix Core.
+Assuming that this user is named "perforce", please make sure that it can run the `p4 info` command.
+
+The simplest way of doing this is by adding the P4PORT setting to the .profile file. For example:
+
+```
+echo "export P4PORT=ssl:1666" >> ~/.profile
+```
+
+Note: The reason we are using the "perforce" user is that by default, it has access to the
+Helix Core metadata, checkpoints and version files. It might be a good idea to use a more restricted
+user as long as it can run p4d commands and access the versioned files and checkpoints.
+
+## Make sure the perforce user can run gcloud
+
+Login as the perforce user (e.g. `sudo -u perforce -i`) and make sure you can run the
+gcloud command. Many issues can be resolved by installing the latest version of gcloud:
+https://cloud.google.com/sdk/docs/install
+
+
+## Copying the backup scripts
+
+Simply place the two backup scripts (nightly_backup.sh and monthly_backup.sh) in any folder
+readable by the perforce user and make sure they are executable (via chmod u+x).
+
+## Setting up the offline root
+
+This backup strategy uses a complete copy of the metadata in offline mode to prevent locking the
+main database during checkpoint operations.
+
+To create an offline root directory:
+
+1. Create a new directory on a disk which has enough storage to hold metadata. Note that the script
+expects the server id to be part of that path. For example, if the server is named "demo", an
+offline root directory could be /opt/perforce/offline:
+
+```
+mkdir -p /opt/perforce/offline/demo
+```
```

Comment:
Do you want to add a chmod line here to remove world-readable access incase of sensitive files.

Reply:
Added chmod 700 /opt/perforce/offline